exporter/bmchelixexporter/exporter.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package bmchelixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/bmchelixexporter"
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// BmcHelixExporter is responsible for exporting metrics to BMC Helix
+type BmcHelixExporter struct {
+	config            *Config
+	logger            *zap.Logger
+	version           string
+	telemetrySettings component.TelemetrySettings
+	metricsProducer   *MetricsProducer
+	metricsClient     *MetricsClient
+}
+
+// newBmcHelixExporter instantiates a new BMC Helix Exporter
+func newBmcHelixExporter(config *Config, createSettings exporter.Settings) (*BmcHelixExporter, error) {
+	if config == nil {
+		return nil, errors.New("nil config")
+	}
+
+	return &BmcHelixExporter{
+		config:            config,
+		version:           createSettings.BuildInfo.Version,
+		logger:            createSettings.Logger,
+		telemetrySettings: createSettings.TelemetrySettings,
+	}, nil
+}
+
+// pushMetrics is invoked by the OpenTelemetry Collector to push metrics to BMC Helix
+func (be *BmcHelixExporter) pushMetrics(ctx context.Context, md pmetric.Metrics) error {
+	be.logger.Info("Building BMC Helix payload")
+	helixMetrics, err := be.metricsProducer.ProduceHelixPayload(md)
+	if err != nil {
+		be.logger.Error("Failed to build BMC Helix payload", zap.Error(err))
+		return err
+	}
+
+	be.logger.Info("Sending BMC Helix payload")
+	err = be.metricsClient.SendHelixPayload(ctx, helixMetrics)
+	if err != nil {
+		be.logger.Error("Failed to send BMC Helix payload", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+// start is invoked during service start
+func (be *BmcHelixExporter) start(ctx context.Context, host component.Host) error {
+	be.logger.Info("Starting BMC Helix Exporter")
+
+	// Get the hostname reported by the kernel
+	osHostname, err := os.Hostname()
```

Comment:
Depending on this starts up, it could fetch the container hostname which is just a hash, is that going to be an issue?

Reply:
Very good question! This shouldn't cause issues for the user, as they should be able to configure the hostname of their container or pod. If that's not feasible, the hash will change with each new deployment. In such cases, I would recommend using OTTL to override the host.name on either the resource or the metric itself, though this approach could become cumbersome in scenarios where the collector processes a large volume of metrics. Alternatively, we could introduce an `entity_hostname` configuration option to override the OS hostname. Currently, the code defaults to the OS hostname only if it cannot find a `host.name` attribute on the datapoint or the associated resource. What do you think, @bertysentry?

Reply:
That's a good catch. I recommend we don't use `os.Hostname()` as the default entity hostname in BMC Helix. Instead, if the user wants to use the local host details, we will recommend to simply use the [resourcedetectionprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md) in their otelcol config.

And if metrics don't have a `host.name` (at the end of the processors chain), then such metrics won't be exported to BMC Helix, and it's okay. Maybe we should log the number of metrics that are dropped so the end user knows what is going on.

Reply:
Ok, `os.Hostname()` will not be used in that case.